src/utility/firebase.js

```diff
@@ -0,0 +1,18 @@
+import firebase from 'firebase/app'
+import 'firebase/firestore'
+
+if (!firebase.apps.length) {
```

Comment:
What is this conditional checking? To see if there's a firebase app?

Reply:
Yeah, this file will be imported many times and we only want a single app

Reply:
Maybe a good place to use a singleton?

Reply:
this is the singleton

Reply:
oh lmao all the singletons i've seen before had a slightly different structure but this works